### ISSUE TITLE
2.5.0/2.5.1 restore from snapshot / init from genesis fail?

### ISSUE THREAD

**Issue by jun0tpyrc** [axieinfinity/ronin#247]
2.5.0 using genesis
```
main.main()
        github.com/ethereum/go-ethereum/cmd/ronin/main.go:279 +0x35
INFO [03-23|08:08:56.075] Maximum peer count                       ETH=50 LES=0 total=50
INFO [03-23|08:08:56.075] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [03-23|08:08:56.076] Set global gas cap                       cap=50,000,000
INFO [03-23|08:08:56.077] Allocated trie memory caches             clean=154.00MiB dirty=256.00MiB
INFO [03-23|08:08:56.077] Allocated cache and file handles         database=/opt/ronin/ronin/chaindata cache=512.00MiB handles=32767
INFO [03-23|08:08:56.116] Opened ancient database                  database=/opt/ronin/ronin/chaindata/ancient readonly=false
INFO [03-23|08:08:56.117] Initialised chain configuration          config="{ChainID: 2020 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Petersburg: 0 Istanbul: 49777
78, Odysseus: 10301597, Fenix: <nil>, Muir Glacier: <nil>, Berlin: <nil>, London: <nil>, Arrow Glacier: <nil>, Engine: consortium, Blacklist Contract: 0x313b24994c93FA0471CB4D7aB796b07467041806, Fenix Validator Contract: <
nil>, ConsortiumV2: <nil>, ConsortiumV2.RoninValidatorSet: 0x0000000000000000000000000000000000000000, ConsortiumV2.SlashIndicator: 0x0000000000000000000000000000000000000000, ConsortiumV2.StakingContract: 0x00000000000000
00000000000000000000000000}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xb6f245]

goroutine 1 [running]:
```

2.5.1 using https://github.com/axieinfinity/ronin-snapshot / genesis
```
INFO [03-23|08:00:06.002] Persisted trie from memory database      nodes=36 size=3.87KiB time="129.602µs" gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=-246.00B
Fatal: Failed to write genesis block: mismatching Fenix fork block in database (have 14938103, want <nil>, rewindto 14938102)
```

**Comment by jun0tpyrc** [comment]
doc  link in https://docs.roninchain.com/docs/intro/mainnet for genesis is old…

please update doc "https://docs.roninchain.com/assets/files/genesis-mainnet-0432c8c2c92e729b3d27005ae7fcb4c4.json" to
https://raw.githubusercontent.com/axieinfinity/ronin/master/genesis/mainnet.json ?

**Comment by jun0tpyrc** [comment]
still can't run
```
INFO [03-23|08:18:39.116] Opened ancient database                  database=/opt/ronin/ronin/chaindata/ancient readonly=false
INFO [03-23|08:18:39.118] Initialised chain configuration          config="{ChainID: 2020 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Petersburg: 0 Istanbul: 49777
78, Odysseus: 10301597, Fenix: 14938103, Muir Glacier: <nil>, Berlin: <nil>, London: <nil>, Arrow Glacier: <nil>, Engine: consortium, Blacklist Contract: 0x313b24994c93FA0471CB4D7aB796b07467041806, Fenix Validator Contract
: 0x7f13232Bdc3a010c3f749a1c25bF99f1C053CE70, ConsortiumV2: <nil>, ConsortiumV2.RoninValidatorSet: 0x0000000000000000000000000000000000000000, ConsortiumV2.SlashIndicator: 0x0000000000000000000000000000000000000000, Consor
tiumV2.StakingContract: 0x0000000000000000000000000000000000000000, Puffy: <nil>, Buba: <nil>}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xb704c5]
```

**Comment by minh-bq** [comment]
Could you try to add this line to genesis to see if it works?
```
    "consortiumV2Block": 11706000000000000000000000000000000,
```

We are planning to release a new version and update the documentation too.

**Comment by jun0tpyrc** [comment]
> 
seem adding that key as `config.consortiumV2Block` can get pass the restart loop for init v2.5.1 with genesis, will retry with data snapshot also 

```
INFO [03-23|08:31:50.985] Initialised chain configuration          config="{ChainID: 2020 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Petersburg: 0 Istanbul: 4977778, Odysseus: 10301597, Fenix: 14938103, Muir Glacier: <nil>, Berlin: <nil>, London: <nil>, Arrow Glacier: <nil>, Engine: consortium, Blacklist Contract: 0x313b24994c93FA0471CB4D7aB796b07467041806, Fenix Validator Contrac$
: 0x7f13232Bdc3a010c3f749a1c25bF99f1C053CE70, ConsortiumV2: 11706000000000000000000000000000000, ConsortiumV2.RoninValidatorSet: 0x0000000000000000000000000000000000000000, ConsortiumV2.SlashIndicator: 0x00000000000000000$
0000000000000000000000, ConsortiumV2.StakingContract: 0x0000000000000000000000000000000000000000, Puffy: <nil>, Buba: <nil>}"
INFO [03-23|08:31:50.986] Initialising Ethereum protocol           network=2020 dbversion=<nil>
INFO [03-23|08:31:50.988] Loaded most recent local header          number=0 hash=6e675e..22a1a4 td=7 age=53y11mo4w
INFO [03-23|08:31:50.988] Loaded most recent local full block      number=0 hash=6e675e..22a1a4 td=7 age=53y11mo4w
INFO [03-23|08:31:50.988] Loaded most recent local fast block      number=0 hash=6e675e..22a1a4 td=7 age=53y11mo4w
WARN [03-23|08:31:50.988] Failed to load snapshot, regenerating    err="missing or corrupted snapshot"
INFO [03-23|08:31:50.989] Rebuilding state snapshot
INFO [03-23|08:31:50.991] Resuming state snapshot generation       root=d7f5c1..421918 accounts=0 slots=0 storage=0.00B elapsed=1.292ms
INFO [03-23|08:31:50.991] Loaded local transaction journal         transactions=0 dropped=0
INFO [03-23|08:31:50.991] Regenerated local transaction journal    transactions=0 accounts=0
INFO [03-23|08:31:50.993] Gasprice oracle is ignoring threshold set threshold=2
WARN [03-23|08:31:50.993] Error reading unclean shutdown markers   error="leveldb: not found"
INFO [03-23|08:31:50.993] Starting peer-to-peer node               instance=ronin/v2.5.1-d1a6cc9b/linux-amd64/go1.17
INFO [03-23|08:31:50.995] Generated state snapshot                 accounts=4 slots=22 storage=1.87KiB elapsed=5.227ms
INFO [03-23|08:31:51.024] New local node record                    seq=1,679,556,807,084 id=b1cc5dded17224de ip=127.0.0.1 udp=30303 tcp=30303
INFO [03-23|08:31:51.025] Started P2P networking                   self=enode://f731f08d2ed57d44fa70a811a9523d7fd06ba12a3d8ca61664120ffbea95bcd3da25665902496b7d52f318ec7b9da682fe0ec835d9cbd36963a2ee7a880a5f11@127.0.0.1:30$
03
INFO [03-23|08:31:51.026] IPC endpoint opened                      url=/opt/ronin/geth.ipc
INFO [03-23|08:31:51.027] HTTP server started                      endpoint=[::]:8545 prefix= cors=* vhosts=*
INFO [03-23|08:31:51.027] WebSocket enabled                        url=ws://[::]:8546
```

**Comment by minh-bq** [comment]
We have released the new version 2.5.2 with the correct genesis
